iceoryx_posh/source/roudi/roudi.cpp

```diff
@@ -358,15 +346,35 @@ void RouDi::processMessage(const runtime::IpcMessage& message,
     }
     case runtime::IpcMessageType::FIND_SERVICE:
     {
-        if (message.getNumberOfElements() != 4)
+        if (message.getNumberOfElements() != 6)
         {
             LogError() << "Wrong number of parameters for \"IpcMessageType::FIND_SERVICE\" from \"" << runtimeName
                        << "\"received!";
         }
         else
         {
-            capro::IdString_t service{cxx::TruncateToCapacity, message.getElementAtIndex(2)};
-            capro::IdString_t instance{cxx::TruncateToCapacity, message.getElementAtIndex(3)};
+            cxx::variant<cxx::Wildcard_t, capro::IdString_t> service;
+            cxx::variant<cxx::Wildcard_t, capro::IdString_t> instance;
+            bool isServiceWildcard = false;
+            bool isInstanceWildcard = false;
+            cxx::convert::fromString(message.getElementAtIndex(2).c_str(), isServiceWildcard);
+            if (!isServiceWildcard)
```

Comment:
I see no need to negate here, drop the `!` and just switch the statements in the `if` and `else` branches. Maybe  the compiler optimizes this away, maybe not, but for readability prefer less complexity if it is not needed.

Reply:
No else branches anymore, would leave it as it is.